### PR TITLE
Add schema for the externalarray type

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

**Comment by nden** [comment]
We try to include a `description` field in all schemas. Could you please add one.
Also, when possible, include examples. Here's an example

https://github.com/spacetelescope/asdf-standard/blob/master/schemas/stsci.edu/asdf/unit/unit-1.0.0.yaml

File: schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - type: string
+  datatype:
+    type: string
+  shape:
+    type: array
+    items:
+      anyOf:
+      - type: integer
+        minimum: 0
+      - enum: ['*']
```
Comment (drdavella) [inline schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml]: It's not clear to me what purpose this serves.

File: schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/core/externalarray-1.0.0"
+tag: "tag:stsci.edu:asdf/core/externalarray-1.0.0"
+title: Point to an array-like object in an external file.
+
```
Comment (drdavella) [inline schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml]: An example would be useful, and also a description, as @nden mentioned.

File: schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml (hunk 1 above)
Comment (Cadair) [reply schemas/stsci.edu/asdf/core/externalarray-1.0.0.yaml]: 🤷‍ I have no idea either. I will remove it and see if it breaks.

**Comment by drdavella** [comment]
@Cadair https://github.com/spacetelescope/asdf/pull/400 seems ready for merge, but this schema could still use a brief description.

**Comment by Cadair** [comment]
🎉 thanks!

**Pull request merged**